File: 백준/Gold/32957. 절대적인 스왑/절대적인 스왑.py
# 절대적인 스왑
import sys
input = sys.stdin.readline

N = int(input())
nums = list(map(int,input().split()))

maxx = 0
minn = 1000001
value= 0

for i in range(1, N + 1):
    num = nums[i-1]
    maxx = max(maxx, min(num, i)) # 각 위치의 작은 값 중 최대값
    minn = min(minn, max(num, i)) # 각 위치의 큰 값 중 최소값
    value += abs(num - i) # 원래의 가치

print(value + 2 * max(0, maxx - minn))

File: 백준/Gold/32957. 절대적인 스왑/절대적인 스왑.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int N = Integer.parseInt(br.readLine());
        int[] nums = new int[N];

        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            nums[i] = Integer.parseInt(st.nextToken());
        }

        int max = 0;
        int min = 1000001;
        long value = 0;

        for (int i = 1; i < N+1; i++) {
            int num = nums[i-1];
            max = Math.max(max, Math.min(num, i));
            min = Math.min(min, Math.max(num, i));

            value += Math.abs(num - i);
        }

        System.out.println(value + (long) 2 * Math.max(0, max - min));
    }
}
